### PR TITLE
Transaction support for actors using cosmos

### PR DESCRIPTION
**Pull request by LMWF**
# Description

These are dapr side changes corresponding to https://github.com/dapr/components-contrib/pull/372.  This PR must go in after the one in components-contrib (go.mod etc needs updates) - it's being opened now to provide context into how the calls from dapr will appear.  Please see https://github.com/dapr/components-contrib/pull/372 for description.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1723 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -347,9 +347,14 @@ func (a *actorsRuntime) GetState(ctx context.Context, req *GetStateRequest) (*St
 	if a.store == nil {
 		return nil, errors.New("actors: state store does not exist or incorrectly configured")
 	}
+
+	partitionKey := a.constructCompositeKey(a.config.AppID, req.ActorType, req.ActorID)
+	metadata := map[string]string{"partitionkey": partitionKey}
```
Comment (youngbupark) [inline pkg/actors/actors.go:352]: This is especially for actors. I think we need to update api or some docs?

File: pkg/actors/actors.go [hunk 2]
```diff
@@ -365,6 +370,9 @@ func (a *actorsRuntime) TransactionalStateOperation(ctx context.Context, req *Tr
 		return errors.New("actors: state store does not exist or incorrectly configured")
 	}
 	requests := []state.TransactionalRequest{}
+	partitionKey := a.constructCompositeKey(a.config.AppID, req.ActorType, req.ActorID)
+	metadata := map[string]string{"partitionkey": partitionKey}
```
Comment (youngbupark) [inline pkg/actors/actors.go:374]: Can we add the test?

File: pkg/actors/actors.go (hunk 1 above)
Comment (LMWF) [reply pkg/actors/actors.go:352]: Yes, this https://github.com/dapr/docs/blob/master/howto/setup-state-store/setup-azure-cosmosdb.md.  I'll update that after this is finalized.

File: pkg/actors/actors.go (hunk 2 above)
Comment (LMWF) [reply pkg/actors/actors.go:374]: Added

**Comment by LMWF** [comment]
Going to open in a different PR.

**Pull request closed without merging**